modules/jpa/src/main/java/io/oasp/module/jpa/dataaccess/base/AbstractGenericDao.java

```diff
@@ -171,13 +174,26 @@ public boolean exists(ID id) {
     CriteriaQuery<E> query = builder.createQuery(getEntityClass());
     Root<E> root = query.from(getEntityClass());
     query.select(root);
-    query.where(root.get("id").in(ids));
+    query.where(root.get("id").in(toCollection(ids)));
     TypedQuery<E> typedQuery = getEntityManager().createQuery(query);
     List<E> resultList = typedQuery.getResultList();
     LOG.debug("Query for selection of {} objects returned {} hit(s).", getEntityName(), resultList.size());
     return resultList;
   }
 
+  /**
+   * @param ids sequence of id
+   * @return a collection of these ids to use {@link Predicate#in(Collection)} for instance
+   */
+  protected Collection<ID> toCollection(Iterable<ID> ids) {
+
+    final Collection<ID> idsList = new ArrayList<>();
```

Comment:
Will technically fix it but is a kind of waste if the `Iterable` already is a `Collection`. IMHO we should first check with instanceof and then only cast to avoid pointless conversion. WDYT?

Reply:
You are absolutely right. I'll change my PR.

Reply:
You are absolutely right. I'll change my PR.